### PR TITLE
Add new navbar and reorganize navigation elements, refs #552

### PR DESCRIPTION
**Pull request by hackmastera**
Still more to do on the navbar, but this is in a mergeable state as we continue making progress.

screen shot 2017-05-25 at 11 31 58 am

### CODE REVIEW

File: app/views/_controls.html.erb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<nav id="searchbar" class="navbar navbar-default navbar-static-top">
+  <div class="container-fluid">
+    <div class="row">
+      <div class="searchbar-right navbar-right col-sm-7 col-md-6">
+        <%= render partial: 'catalog/search_form' %>
+      </div>
+    </div>
+  </div>
+</nav><!-- /.navbar -->
```
Comment (jrochkind) [inline app/views/_controls.html.erb]: when overriding a partial from sufia/CC (I think this is?), what do you think about leaving a comment with path/version it was overridden from (github 'y' url works nicely) and why? Not sure which partials in this PR are ours, and which are overridden.

File: app/views/_masthead.html.erb [hunk 2]
```diff
@@ -0,0 +1,37 @@
+<% if current_ability.current_user.staff? %>
+  <%= render '/shared/select_work_type_modal' if create_work_presenter.many? %>
+  <nav id="staff-navbar" class="navbar navbar-inverse navbar-static-top staff-navbar" role="navigation">
+    <div class="container-fluid">
+      <div class="collapse navbar-collapse" id="staff-navbar-collapse">
+        <%# TODO: make these collapse to hamburger on mobile? two hamburgers is weird but it takes up lots of space otherwise %>
```
Comment (jrochkind) [inline app/views/_masthead.html.erb]: i don't know if this is your comment or sufia's, but I'm not worried about admin-only stuff on mobile, admin won't be working on mobile much/at all I think.

File: app/views/_controls.html.erb (hunk 1 above)
Comment (hackmastera) [inline app/views/_controls.html.erb]: Yes will do that thanks for the reminder. both of these are overrides.

File: app/views/_masthead.html.erb (hunk 2 above)
Comment (hackmastera) [inline app/views/_masthead.html.erb]: my comment. I'm fine with this approach / assumption!

**Comment by jrochkind** [comment]
okay, I'm not _totally_ sure about all this stuff, but let's go with it!

**Pull request merged**